src/main/java/com/minecolonies/coremod/colony/TriggerColonyAchievements.java

```diff
@@ -0,0 +1,232 @@
+package com.minecolonies.coremod.colony;
+
+import com.minecolonies.coremod.achievements.ModAchievements;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Trigger the corresponding colony achievement.
+ */
+public class TriggerColonyAchievements
+{
+    private static final String TAG_LUMBERJACK_SAPLINGS   = "saplings";
+    private static final String TAG_LUMBERJACK_TREES      = "trees";
+    private static final String TAG_FISHERMAN_FISH        = "fish";
+    private static final String TAG_BUILDER_HUTS          = "huts";
+    private static final String TAG_GUARD_MOBS            = "mobs";
+    private static final String TAG_FARMER_CARROTS        = "carrots";
+    private static final String TAG_FARMER_POTATOES       = "potatoes";
+    private static final String TAG_FARMER_WHEAT          = "wheat";
+    private static final String TAG_MINER_DIAMONDS        = "diamonds";
+    private static final String TAG_MINER_ORES            = "ores";
+
+    /**
+     * Trigger first achievement.
+     * @param statistic the statistic.
+     */
+    public static void triggerFirstAchievement(@NotNull String statistic, @NotNull Colony colony)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "triggerFirstAchievement" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document the parameter(s): colony [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)